kernel/byterun/coq_values.c

```diff
@@ -96,3 +96,14 @@ value coq_tcode_array(value tcodes) {
   }
   CAMLreturn(res);
 }
+
+CAMLprim value coq_obj_set_tag (value arg, value new_tag)
+{
+#if OCAML_VERSION >= 50000
+// Noop placeholder used by native_compute
```

Comment:
Raising exceptions from C code is more annoying, I'd rather just keep it as-is.

Reply:
What does "annoying" mean here?

Reply:
Like you have to include an additional header and register a static string. I'm not even sure if there is an easy way to raise the Assertion_failure error without registering it from the OCaml world, since OCaml C assert calls caml_failed_assert only in debug mode and then you have to pass it weird stuff.

Reply:
I agree that we should assert false here. But there is no need to do it the OCaml way. Just do it the C way, either `assert(false)` or `abort()`. It is strictly better than silently ignoring the issue.